[All Hosts] (manifest) update requirements-related docs (#3190)
OfficeDev/office-js-docs-pr: docs/develop/specify-office-hosts-and-api-requirements.md

     </Methods>
  </Requirements>
    ...
</OfficeApp>
```
Note the following about this example:

- The **Requirements** element contains the **Sets** and **Methods** child elements.
- The **Sets** element can contain one or more **Set** elements. `DefaultMinVersion` specifies the default `MinVersion` value of all child **Set** elements.
- A [Set](../reference/manifest/set.md) element specifies a requirement set that the Office application must support to make the add-in installable. The `Name` attribute specifies the name of the requirement set. The `MinVersion` specifies the minimum version of the requirement set. `MinVersion` overrides the value of the `DefaultMinVersion` attribute in the parent **Sets**.
- The **Methods** element can contain one or more [Method](../reference/manifest/method.md) elements. You can't use the **Methods** element with Outlook add-ins.
- The **Method** element specifies an individual method that the Office application must support to make the add-in installable. The `Name` attribute is required and specifies the name of the method qualified with its parent object.

## Design for alternate experiences

> The extensibility features that the Office add-in platform provides can be usefully divided into three kinds:

Author: I want to keep my wording here. There are MANY ways to divide and categorize extensibility features. I want to emphasize that this is only one way. Moreover, this way of dividing into 3 is only relevant to this article. I don't want to imply that there is some grander, transcendent significance to this particular division.

> - Extensibility features that are available immediately after the add-in is installed. You can make use of this kind of feature by configuring a [VersionOverrides](../reference/manifest/versionoverrides.md) element in the manifest. An example of this kind of feature is [Add-in Commands](../design/add-in-commands.md), which are custom ribbon buttons and menus.

Author: I think this change would create an incomplete sentence.

> - Extensibility features that are available only when the add-in is running and that are implemented with Office.js JavaScript APIs; for example, [Dialog Boxes](../design/dialog-boxes.md).

Author: This change creates an incomplete sentence.

> - Extensibility features that are available only at runtime but are implemented with a combination of Office.js JavaScript and configuration in a **VersionOverrides** element. Examples of these are [Excel custom functions](../excel/custom-functions-overview.md), [single sign-on](sso-in-office-add-ins.md), and [custom contextual tabs](../design/contextual-tabs.md).

Author: This change would create an incomplete sentence.

| Workbook      | Excel on the web, Windows, Mac, iPad           | Task pane, Content     |

> [!NOTE]
> Office applications are supported on different platforms and run on desktops, web browsers, tablets, and mobile devices. You usually can't specify which platform can be used to run your add-in. For example, if you specify `Workbook`, both Excel on the web and on Windows can be used to run your add-in. However, if you specify `Mailbox`, your add-in won't run on Outlook mobile clients unless you define the [mobile extension point](../reference/manifest/extensionpoint.md#mobilemessagereadcommandsurface).

> [!NOTE]

> > It is not possible for an add-in manifest to apply to more than one type: Mail, Task pane, or Content. This means that if you want your add-in to be installable on Outlook and on one of the other Office applications, you must create *two* add-ins, one with a Mail type manifest and the other with a Task pane or Content type manifest.

Reviewer: ```suggestion
> It isn't possible for an add-in manifest to apply to more than one type: Mail, Task pane, or Content. This means that if you want your add-in to be installable on Outlook and on one of the other Office applications, you must create *two* add-ins, one with a Mail type manifest and the other with a Task pane or Content type manifest.
```

## Design for alternate experiences

The extensibility features that the Office add-in platform provides can be usefully divided into three kinds:

- Extensibility features that are available immediately after the add-in is installed. You can make use of this kind of feature by configuring a [VersionOverrides](../reference/manifest/versionoverrides.md) element in the manifest. An example of this kind of feature is [Add-in Commands](../design/add-in-commands.md), which are custom ribbon buttons and menus.
- Extensibility features that are available only when the add-in is running and that are implemented with Office.js JavaScript APIs; for example, [Dialog Boxes](../design/dialog-boxes.md).
- Extensibility features that are available only at runtime but are implemented with a combination of Office.js JavaScript and configuration in a **VersionOverrides** element. Examples of these are [Excel custom functions](../excel/custom-functions-overview.md), [single sign-on](sso-in-office-add-ins.md), and [custom contextual tabs](../design/contextual-tabs.md).

> If your add-in uses a specific extensibility feature for some of its functionality but has other useful functionality that doesn't require the extensibility feature, you should design the add-in so that it is installable on platform and Office version combinations that don't support the extensibility feature. It can provide a valuable, albeit diminished, experience on those combinations. 

Reviewer: ```suggestion
If your add-in uses a specific extensibility feature for some of its functionality but has other useful functionality that doesn't require the extensibility feature, you should design the add-in so that it's installable on platform and Office version combinations that don't support the extensibility feature. It can provide a valuable, albeit diminished, experience on those combinations. 
```

- The second parameter is optional. It's a string that specifies the minimum requirement set version that the Office application must support in order for the code within the `if` statement to run (e.g., "**1.9**"). If not used, version "1.1" is assumed.

> [!WARNING]
> When calling the `isSetSupported` method, the value of the second parameter (if specified) should be a string not a number. The JavaScript parser cannot differentiate between numeric values such as 1.1 and 1.10, whereas it can for string values such as "1.1" and "1.10".

The following table shows the requirement set names for the application specific API models.

|Office application|RequirementSetName|
|---|---|
|Excel|ExcelApi|
|OneNote|OneNoteApi|
|Outlook|Mailbox|
|PowerPoint|PowerPointApi|
|Word|WordApi|

The following is an example of using the method with one of the Common API model requirement sets.

```js
if (Office.context.requirements.isSetSupported('CustomXmlParts'))
{
    // Run code that uses API members from the CustomXmlParts requirement set.
}
else
{

>     // Run alternate code when the user's Word does not support the CustomXmlParts requirement set.

Reviewer: ```suggestion
    // Run alternate code when the user's Word doesn't support the CustomXmlParts requirement set.
```

}
```

### Specifying requirements in a VersionOverrides element

The [VersionOverrides](../reference/manifest/versionoverrides.md) element was added to the manifest schema primarily, but not exclusively, to support features that must be available immediately after an add-in is installed, such as add-in commands (custom ribbon buttons and menus). Office must know about these features when it parses the add-in manifest. 

> When your add-in uses one of these features, but the add-in is valuable, and should be installable, even on Office versions that don't support the feature, you can identify the feature using a [Requirements](../reference/manifest/requirements.md) element (and its child [Sets](../reference/manifest/sets.md) and [Methods](../reference/manifest/methods.md) elements) *as a child of the **VersionOverrides** element itself instead of in the base manifest (that is, as a child of the `OfficeApp` element*). The effect of doing this is that Office will allow the the add-in to be installed, but Office will ignore certain of the child elements of the **VersionOverrides** element on Office versions where the feature is not supported. 

Author: ```suggestion
Suppose your add-in uses one of these features, but the add-in is valuable, and should be installable, even on Office versions that don't support the feature. In this scenario, identify the feature using a [Requirements](../reference/manifest/requirements.md) element (and its child [Sets](../reference/manifest/sets.md) and [Methods](../reference/manifest/methods.md) elements) that you include as a child of the **VersionOverrides** element itself instead of in the base manifest (that is, as a child of the `OfficeApp` element). The effect of doing this is that Office will allow the add-in to be installed, but Office will ignore certain of the child elements of the **VersionOverrides** element on Office versions where the feature is not supported. 
```
